### PR TITLE
EN-2979 support sensitive pages

### PR DESCRIPTION
**Pull request by nitzanpx**

### CODE REVIEW

File: src/Perimeterx.php [hunk 1]
```diff
@@ -96,6 +96,7 @@ private function __construct(array $pxConfig = [])
                 'api_timeout' => 1,
                 'api_connect_timeout' => 1,
                 'local_proxy' => false,
+                'sensetive_routes_prefix' => []
```
Comment (yaronpx) [inline src/Perimeterx.php]: *sensitive.
This typo appears several times in this PR.

File: src/PerimeterxContext.php [hunk 2]
```diff
@@ -399,6 +400,15 @@ public function setDecodedCookie($cookie)
         $this->decoded_px_cookie = $cookie;
     }
 
+
+    private function isSensitiveRoutePrefix($sensetive_routes_prefix, $uri){
+      foreach($sensetive_routes_prefix as $route){
+        if (strpos($uri, $route) === 0){
```
Comment (barakpx) [inline src/PerimeterxContext.php]: consider using strncmp with the route length - in this case we will not scan the full uri to know that the prefix is not there.

File: src/PerimeterxContext.php [hunk 3]
```diff
@@ -399,6 +400,15 @@ public function setDecodedCookie($cookie)
...
+          return true;
+        }
+      }
```
Comment (barakpx) [inline src/PerimeterxContext.php]: return false; just to be align with types.

File: tests/PerimeterxCookieValidatorTest.php [hunk 4]
```diff
@@ -275,6 +275,38 @@ public function testCookieHmacValid() {
...
+        $cookie_hmac = '826bb9324795dd2e621e133372a627f7b5d9523978fd1c3337389b9fa1f5cbc7';
+        $cookie_score_a = 0;
+        $cookie_score_b = 0;
+
+        $pxCookie = $this->encodeCookie(
+                $this->createCookie($cookie_time, $cookie_vid, $cookie_uuid, $cookie_hmac, $cookie_score_a, $cookie_score_b)
+            );
+        $userAgent = self::USER_AGENT;
+        $ip = self::IP;
+        $pxCtx = $this->getPxContext($pxCookie, $userAgent, $ip, true);
```
Comment (barakpx) [inline tests/PerimeterxCookieValidatorTest.php]: don't get this one - we initialize the context with the desired state without testing how configure a path cause risk request to be considered sensitive

**Pull request merged**